website/docs/plugin/framework/functions/errors.mdx

```diff
@@ -0,0 +1,149 @@
+---
+page_title: 'Plugin Development - Framework: Function Errors'
+description: |-
+  How to return function errors from the Terraform provider development
+  framework.
+---
+
+# Returning Function Errors
+
+Providers use [`FuncError`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/function#FuncError) to
+surface a practitioner-facing error generated during execution of provider-defined functions. These errors are
+returned from Terraform CLI at the end of command output:
+
+```console
+$ terraform plan
+# ... other plan output ...
+╷
+│ Error: Message
+│ 
+│   on example.tf line #:
+│    #: source configuration line
+╵
+```
+
+In the framework, you may encounter them in response structs or as returns from
+provider-defined function execution.:
+
+```go
+func (f *ExampleFunction) Run(ctx context.Context, req function.RunRequest, resp *function.RunResponse) {
+```
+
+This is the most common form for `FuncError`: a single error whose text
+is the concatenated error text from one or more errors. This approach allows
+your provider to inform practitioners about all relevant errors at the same
+time, allowing practitioners to fix their configuration or environment more
+quickly. You should only concatenate a `FunctError` and never replace or
+remove information it.
+
+The next section will detail the concepts and typical behaviors of
+function error, while the final section will outline the typical methods for
+working with function error, using functionality from the available
+[`function` package](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/function).
+
+## Function Error Concepts
+
+### Text
+
+`Text` is a practitioner-oriented description of the problem. This should
+contain sufficient detail to provide both general and more specific information
+regarding the issue. For example "Error executing function: foo can only contain
+letters, numbers, and digits."
+
+### FunctionArgument
+
+`FunctionArgument` identifies the specific function argument position that caused the
+error. Only errors that pertain to a function argument will include this information.
```

Comment:
nit: Maybe add a note that it is an `int64` type and that the argument position is zero-based

Reply:
Updated to include your suggestion.